src/Analyser/NodeScopeResolver.php

```diff
@@ -4522,8 +4524,22 @@ private function processArgs(
 			}
 
 			if ($assignByReference) {
+				if ($currentParameter === null) {
+					throw new ShouldNotHappenException();
+				}
+
 				$argValue = $arg->value;
 				if ($argValue instanceof Variable && is_string($argValue->name)) {
+
+					$paramOutTypes = $this->getParameterOutExtensionsTypes($callLike, $calleeReflection, $currentParameter, $scope);
```

Comment:
Will this realistically ever return multiple types? I feel like the union should be done inside this method, and the method should return `?Type`.